src/Illuminate/Container/BoundMethod.php

```diff
@@ -105,18 +105,25 @@ protected static function normalizeMethod($callback)
      *
      * @param  \Illuminate\Container\Container  $container
      * @param  callable|string  $callback
-     * @param  array  $parameters
+     * @param array $inputData
      * @return array
+     * @throws \ReflectionException
      */
-    protected static function getMethodDependencies($container, $callback, array $parameters = [])
+    protected static function getMethodDependencies($container, $callback, array $inputData = []) : array
     {
-        $dependencies = [];
+        $signature = static::getCallReflector($callback)->getParameters();
 
-        foreach (static::getCallReflector($callback)->getParameters() as $parameter) {
-            static::addDependencyForCallParameter($container, $parameter, $parameters, $dependencies);
+        // In case the method has no explicit input parameters defined
+        // we will call it, with whatever input data available to us.
+        if (count($signature) === 0) {
+            return $inputData;
         }
 
-        return array_merge($dependencies, $parameters);
+        if (\Illuminate\Support\Arr::isAssoc($inputData)) {
```

Comment:
You should import this at the top, and just use `Arr` here.

Reply:
I have no idea about the standard conventions of laravel for it.
It can also be imported.